mitodl/mit-open
# Adjust OfferedBy and Platform values

# What are the relevant tickets?
Closes #9 (mostly - care will need to be taken to assign correct platform and offered_by values when completing #102)

# Description (What does it do?)
- Modifies the valid `LearningResourceOfferor` and `LearningPlatform` objects
- Adds platform as a filterable api query parameter
- Changes `offered_by` to a single foreign key relation on `LearningResource`


# How can this be tested?
- Run the migrations (`docker compose up` or `docker compose run --rm web ./manage.py migrate`)
- Run `./manage.py update_platforms` and `./manage.py update_offered_by`
- Run `./manage.py backpopulate_xpro_data --delete` and `./manage.py backpopulate_podcast_data --delete`
- Run `./manage.py backpopulate_xpro_data` and `./manage.py backpopulate_podcast_data`
- All xpro courses and programs should now be offered by  `xPRO`, and a mix of `platform` values, one per resource: `emeritus`, `xpro`, `globalalumni`, `whu`, etc
- All podcasts and episodes should have platform `podcast` and various `offered_by` values: `None`, `CSAIL`, etc.
- The homepage courses carousel should display `Offered By - xPRO` for each resource.

You should be able to filter the API by platform and offered_by:

http://localhost:8063/api/v1/learning_resources/?platform=emeritus
http://localhost:8063/api/v1/learning_resources/?offered_by=xpro
